### PR TITLE
feat: add kong integrated with keycloak and with support for many realms

### PR DESCRIPTION
**Pull request by mprevide**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
Add example how using these plugins with keycloak and change the way to get the plugin jwt-keycloak.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR? (Link to an issue, e.g. #99999)** 
#1986

* **Other information**:
The plugin was created by @canattofilipe  and was available at  https://github.com/dojot/kong/tree/feature/keycloak-10.0.1, there are just changes in the way to get the  jwt-keycloak plugin.

### CODE REVIEW

File: api-gateway/kong/examples/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+Clients in the right sidebar:
+
+- Create a new `client` with the *Client ID* as`kong`
+  - Enable the `Enable` option if it is not enabled
+  - Set `Client Protocol` to`openid-connect`
+  - Set `Access Type` to`confidential`
+  - Enable the `Standard Flow Enabled` option if it is not enabled
+  - Disable the `Implicit Flow Enabled` option if it is enabled
+  - Enable the `Direct Access Grants Enabled` option if it is not enabled
+  - Enable the `Service Accounts Enabled` option if it is not enabled (Check if it is necessary)
```
Comment (mprevide) [inline api-gateway/kong/examples/README.md:123]: I need to  remove these `(Check if it is necessary)` and similar

File: api-gateway/kong/examples/README.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  - Go to the `Credentials` tab
+    - Define a password and confirm it in `Password` and`Password Confirmation`, in the example the password `user` is used (Only for local testing)
+    - Disable the `Temporary` option (for local testing only)
+  - Go to the `Role Mappings` tab
+    - In `Available Roles` select `admin` and `Add selected`, so that `admin` will be listed in `Associated Roles`.
+
+- I created a new user with username `user` that will have *role* `user`
+  - Enable the `Email Verified` option (for local testing only)
+  - Go to the `Credentials` tab
+    - Define a password and confirm it in `Password` and`Password Confirmation`, in the example the password `admin` is used (Only for local testing)
```
Comment (thiagodpf) [inline api-gateway/kong/examples/README.md:110]: it seems to me that `passwords` are reversed in relation to the `usernames`.

File: api-gateway/kong/examples/README.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+- Create a new Role with the name 'user'
+
+- Create a new Role with the name 'admin'
+  - Enable the `Composite Roles` option
+  - In `Available Roles` select `user` and `Add selected`, so that `user` will be listed in `Associated Roles`.
+
+### User settings depends on ‘Role settings’
+
+Users in the right sidebar:
```
Comment (thiagodpf) [inline api-gateway/kong/examples/README.md:97]: Wouldn't it be on the _left_ sidebar?

File: api-gateway/kong/examples/README.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  - Enable the `Service Accounts Enabled` option if it is not enabled
+  - Enable the option `Authorization Enabled` if it is not enabled
+  - Set `Valid Redirect URIs` to` http://localhost:8000/* `
+
+#### Configuring authorization
+
+- Go to the `Authorization` tab
+  - Within `Authorization` select the`Authorization Scopes` tab
+    - Create the following scopes: `create`,`delete`, `update` and`view`.
+      - Note: **`create`  is for HTTP**POST**,  `delete` is for HTTP**DELETE**,    `update`is for HTTP **PUT** or **PATCH** e `view` is for HTTP **GET**.
```
Comment (thiagodpf) [inline api-gateway/kong/examples/README.md:135]: > Note: create is for HTTP POST, delete is for HTTP DELETE, update is for HTTP PUT or PATCH **and** view is for HTTP GET

e = and

File: api-gateway/kong/examples/README.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+  - Within `Authorization` select the`Polices` tab
+    - Create a policy for role `admin`
+      - In `Create Policy` select`role`
+        - Set to name `Should be admin
```
Comment (thiagodpf) [inline api-gateway/kong/examples/README.md:139]: Is the formatting incomplete?

File: api-gateway/kong/plugins/pepkong/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+```bash
+luarocks make
+```
+
+# Changing default resource server
+
+To invoke Keycloak authorization service is necessary define a resource server (client), the default value is `kong`,
+but is possible to change it through `CLIENT_ID` environment variable.
+
```
Comment (thiagodpf) [inline api-gateway/kong/plugins/pepkong/README.md:46]: It seems a little confusing to define the "resource server" as "client". I think the _resource server_ is the microservice behind API Gateway and the gateway it is in fact a Keycloak client. 
A more detailed explanation may be needed here...

**Review by thiagodpf — APPROVED** [review]

File: api-gateway/kong/examples/README.md (hunk 3 above)
Comment (mprevide) [reply api-gateway/kong/examples/README.md:97]: Yes =O
Thx

**Review by thiagodpf — DISMISSED** [review]

File: api-gateway/kong/examples/README.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        - Set to name `Should be admin`
+        - Set the `admin` value in`Realm Roles` and select it as `Required`
+        - Set `Logic` to` Positive
+    - Create a policy for role `user`
+      - In `Create Policy` select`role`
+        - Set to name `Should be user
+        - Set the `user` value in`Realm Roles` and select it as `Required`
+        - Set `Logic` to`Positive`
+
+In the following steps It will configure a resource called `server-api-example-sec` in which a user with the role `admin` can create, edit and delete, while a user with the role user can only view:
```
Comment (mprevide) [inline api-gateway/kong/examples/README.md:147]: TODO: Same as passed to `config.resource`  when  add plugin `pepkong` to a service.

**Review by thiagodpf — APPROVED** [review]

**Review by thiagodpf — APPROVED** [review]

**Pull request merged**